src/parser/postgresparser.cpp

```diff
@@ -411,18 +405,19 @@ std::unique_ptr<AbstractExpression> PostgresParser::AExprTransform(A_Expr *root)
   }
 
   ExpressionType target_type;
+  std::vector<std::shared_ptr<AbstractExpression>> children;
 
   if (root->kind == AEXPR_DISTINCT) {
     target_type = ExpressionType::COMPARE_IS_DISTINCT_FROM;
+  } else if (root->kind == AEXPR_OP && root->type == T_TypeCast) {
+    target_type = ExpressionType::OPERATOR_CAST;
   } else {
     auto name = (reinterpret_cast<value *>(root->name->head->data.ptr_value))->val.str;
     target_type = StringToExpressionType(name);
+    children.emplace_back(ExprTransform(root->lexpr));
+    children.emplace_back(ExprTransform(root->rexpr));
   }
 
-  std::vector<std::shared_ptr<AbstractExpression>> children;
-  children.emplace_back(ExprTransform(root->lexpr));
-  children.emplace_back(ExprTransform(root->rexpr));
-
```

Comment:
Do we need to look at root-lexpr and root->rexpr when root->kind is AEXPR_DISTINCT?

Reply:
AEXPR_DISTINCT does seem to use root->lexpr and root->rexpr.
I'll push a change to and fix that.

Reply:
Okay, thanks. Ready to merge once that's in.